src/test/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountEntityTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.pay.connector.gatewayaccount.model;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.pay.connector.gatewayaccountcredentials.model.GatewayAccountCredentialState;
+import uk.gov.pay.connector.gatewayaccountcredentials.model.GatewayAccountCredentialsEntity;
+
+import javax.ws.rs.WebApplicationException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+class GatewayAccountEntityTest {
+    
+    private GatewayAccountEntity gatewayAccountEntity;
+    private List<GatewayAccountCredentialsEntity> gatewayAccountCredentialsEntities = new ArrayList<>();
+    
+    @BeforeEach
+    void setUp() {
+        gatewayAccountEntity = GatewayAccountEntityFixture
+                .aGatewayAccountEntity()
+                .withGatewayAccountCredentials(gatewayAccountCredentialsEntities)
+                .build();
+    }
+
+    @Test
+    void shouldReturnSandboxAsPaymentProviderNameForSingleGatewayAccountCredentialEntity() {
+        gatewayAccountCredentialsEntities.add(new GatewayAccountCredentialsEntity(
+                gatewayAccountEntity,
+                "sandbox",
+                Map.of(),
+                GatewayAccountCredentialState.ACTIVE
+        ));
+
+        assertThat(gatewayAccountEntity.getGatewayNameFromGatewayAccountCredentials(), is("sandbox"));
+    }
+
+    @Test
+    void shouldReturnStripeAsPaymentProviderNameForLatestGatewayAccountCredentialEntity() {
+        GatewayAccountCredentialsEntity latestActiveGatewayAccountCredential = new GatewayAccountCredentialsEntity(
```

Comment:
Can we use fixtures which comes with most of the values as defaults ? 
```suggestion
        GatewayAccountCredentialsEntity credentialsEntity = aGatewayAccountCredentialsEntity()..withPaymentProvider("stripe").build();
```

Reply:
Done.